### PR TITLE
OP-TEE benchmark

### PR DESCRIPTION
**Pull request by igoropaniuk**
Related PRs:

Benchmark app:
https://github.com/linaro-swg/optee_benchmark/pull/1

Changes in OP-TEE
https://github.com/linaro-swg/linux/pull/38
https://github.com/OP-TEE/build/pull/124
https://github.com/OP-TEE/optee_os/pull/1365
https://github.com/OP-TEE/optee_client/pull/79

### CODE REVIEW

**Comment by jbech-linaro** [comment]
This is good, later on we should add the same to the other `*.mk` files also, i.e., not only just QEMU.
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by igoropaniuk** [comment]
Tag applied and rebased on master

File: common.mk [hunk 1]
```diff
@@ -15,12 +15,16 @@ OPTEE_OS_PATH			?= $(ROOT)/optee_os
...
 OPTEE_TEST_PATH			?= $(ROOT)/optee_test
-OPTEE_TEST_OUT_PATH 		?= $(ROOT)/optee_test/out
+OPTEE_TEST_OUT_PATH		?= $(ROOT)/optee_test/out
 HELLOWORLD_PATH			?= $(ROOT)/hello_world
+BENCHMARK_APP_PATH		?= $(ROOT)/optee_benchmark
 
 # default high verbosity. slow uarts shall specify lower if prefered
 CFG_TEE_CORE_LOG_LEVEL		?= 3
 
+# enable latency benchmarks (over all OP-TEE layers)
```
Comment (etienne-lms) [inline common.mk]: s/enable/default disable/

File: common.mk [hunk 2]
```diff
@@ -153,7 +161,7 @@ linux-common: linux-defconfig
 $(LINUX_PATH)/.config: $(LINUX_DEFCONFIG_COMMON_FILES)
 	cd $(LINUX_PATH) && \
 		ARCH=$(LINUX_DEFCONFIG_COMMON_ARCH) \
-		scripts/kconfig/merge_config.sh $(LINUX_DEFCONFIG_COMMON_FILES)
+		scripts/kconfig/merge_config.sh $(LINUX_DEFCONFIG_COMMON_FILES) $(LINUX_DEFCONFIG_BENCH)
```
Comment (etienne-lms) [inline common.mk]: minor:
```Makefile
		scripts/kconfig/merge_config.sh $(LINUX_DEFCONFIG_COMMON_FILES) \
						$(LINUX_DEFCONFIG_BENCH)
```

File: common.mk [hunk 3]
```diff
@@ -250,17 +258,20 @@ OPTEE_OS_COMMON_FLAGS ?= \
...
 OPTEE_OS_CLEAN_COMMON_FLAGS ?= $(OPTEE_OS_COMMON_EXTRA_FLAGS)
 
-optee-os-clean-common: xtest-clean helloworld-clean
+optee-os-clean-common: xtest-clean helloworld-clean benchmark-app-clean-common
 	$(MAKE) -C $(OPTEE_OS_PATH) $(OPTEE_OS_CLEAN_COMMON_FLAGS) clean
 
-OPTEE_CLIENT_COMMON_FLAGS ?= CROSS_COMPILE=$(CROSS_COMPILE_NS_USER)
+OPTEE_CLIENT_COMMON_FLAGS ?= CROSS_COMPILE=$(CROSS_COMPILE_NS_USER) \
+	CFG_TEE_BENCHMARK=$(CFG_TEE_BENCHMARK) \
+	CFG_TEE_PRINT_LATENCY_STAT=$(CFG_TEE_PRINT_LATENCY_STAT)
```
Comment (etienne-lms) [inline common.mk]: No default value for `CFG_TEE_PRINT_LATENCY_STAT`.

File: common.mk [hunk 4]
```diff
@@ -313,6 +325,19 @@ helloworld-clean-common:
 	$(MAKE) -C $(HELLOWORLD_PATH) $(HELLOWORLD_CLEAN_COMMON_FLAGS) clean
 
 ################################################################################
+# benchmark_app
+################################################################################
+BENCHMARK_APP_COMMON_FLAGS ?= HOST_CROSS_COMPILE=$(CROSS_COMPILE_NS_USER)\
```
Comment (etienne-lms) [inline common.mk]: minor: add a space before `\`.

**Comment by igoropaniuk** [comment]
@jbech-linaro 
I hope I may still apply your R-b tag after last additional changes

File: common.mk (hunk 4 above)
Comment (igoropaniuk) [inline common.mk]: fixed

File: common.mk [hunk 5]
```diff
@@ -153,7 +161,8 @@ linux-common: linux-defconfig
 $(LINUX_PATH)/.config: $(LINUX_DEFCONFIG_COMMON_FILES)
 	cd $(LINUX_PATH) && \
 		ARCH=$(LINUX_DEFCONFIG_COMMON_ARCH) \
-		scripts/kconfig/merge_config.sh $(LINUX_DEFCONFIG_COMMON_FILES)
+		scripts/kconfig/merge_config.sh $(LINUX_DEFCONFIG_COMMON_FILES) \
+										$(LINUX_DEFCONFIG_BENCH)
```
Comment (etienne-lms) [inline common.mk]: minor style: too many tabs

Comment (igoropaniuk) [inline common.mk]: fixed

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by igoropaniuk** [comment]
Tags applied and rebased
@jbech-linaro @etienne-lms thanks a lot for your comments!

**Pull request merged**